### ISSUE TITLE
onUploadProgress & onDownloadProgress for node

### ISSUE THREAD

**Issue by alxhotel** [axios/axios#1793]
It would be useful to have the feature of `onUploadProgress` and `onDownloadProgress` also in NodeJS.

Example:

```js
const fs = require('fs');
const axios = require('axios');

var data = fs.readFileSync('test.txt');
axios.post('https://example.com/upload', data, {
  onUploadProgress: function (progressEvent) {
    console.log(Math.floor((progressEvent.loaded * 100) / progressEvent.total));
  }
}).then(function (response) {
  console.log(response);
}).catch(function (error) {
  console.log(error);
});
```

**Comment by RikkiGibson** [comment]
I wrote an adapter for axios to provide uniform behavior for these handlers among other things. You may benefit from seeing the code for it:

https://github.com/Azure/ms-rest-js/blob/master/lib/axiosHttpClient.ts#L103

BTW, your code may work better with [fs.createReadStream](https://nodejs.org/api/fs.html#fs_fs_createreadstream_path_options) as you won't have to wait for the file to be completely read before starting to upload it.

**Comment by RikkiGibson** [comment]
That's right. It's not perfectly accurate, but convenient to implement. I think the discrepancy between the amount of file content read and the amount of content received so far by the server will be at most the `socket.highWaterMark` (defaults to 16 KB).

In case you find it interesting, node.js has built in [backpressure mechanisms](https://nodejs.org/en/docs/guides/backpressuring-in-streams/) to prevent the system from running out of memory e.g. when the disk is faster than the network on a large upload, or the network is faster than the disk on a large download.

**Comment by alxhotel** [comment]
Ah got it. Still in my case the differences is seems to be bigger than 16KB. I guess there isn't an easy solution for this.

**Comment by jacargentina** [comment]
@chinesedfan  any advance on this ?

**Comment by jasonsaayman** [comment]
Please check if this has been fixed with #3694

**Comment by marek-siemieniuk-morawski** [comment]
Hi @jasonsaayman , I've checked and the problem still occurs.
